platform/logging/appenders/Appenders.ts

```diff
@@ -0,0 +1,62 @@
+import { assertNever } from '../../lib/utils';
+import { Schema } from '../../types';
+import {
+  ConsoleAppender,
+  ConsoleAppenderConfigType
+} from './console/ConsoleAppender';
+import { FileAppender, FileAppenderConfigType } from './file/FileAppender';
+import { LogRecord } from '../LogRecord';
+import { Layouts } from '../layouts/Layouts';
+
+/** @internal */
+export type AppenderConfigType =
+  | ConsoleAppenderConfigType
+  | FileAppenderConfigType;
+
+/**
+ * Interface that describes entity that can append `LogRecord` instances to file, stdout, memory or
+ * whatever is implemented internally.
+ * @internal
+ */
+export interface Appender {
+  append(record: LogRecord): void;
+}
+
+/**
+ * Interface that should be additionally implemented by the `Appender`'s when they should
+ * be properly disposed.
+ * @internal
+ */
+export interface DisposableAppender extends Appender {
```

Comment:
The reason is that I don't want Logger to know that its internal appenders can be disposed, only factory should know and control that. What do you think?